backend/app/tests/utils/utils.py

```diff
@@ -39,16 +42,43 @@ def get_superuser_token_headers(client: TestClient) -> dict[str, str]:
     return headers
 
 
-def get_user_id_by_email(db: Session):
+def get_user_id_by_email(db: Session) -> int:
     user = get_user_by_email(session=db, email=settings.FIRST_SUPERUSER)
     return user.id
 
 
+def get_real_api_key_headers(db: Session) -> dict[str, str]:
+    owner_id = get_user_id_by_email(db)
+
+    # Step 1: Create real organization and project
+    organization = Organization(name=f"Test Org {uuid4()}")
+    db.add(organization)
+    db.commit()
+    db.refresh(organization)
+
+    project = Project(name=f"Test Project {uuid4()}", organization_id=organization.id)
+    db.add(project)
+    db.commit()
+    db.refresh(project)
+
+    # Step 2: Create API key
+    api_key = create_api_key(
+        db,
+        organization_id=organization.id,
+        user_id=owner_id,
+        project_id=project.id,
+    )
+
+    return {"X-API-Key": api_key.key}
+
```

Comment:
instead of this can we use seed data? refer this 
`backend/app/tests/api/routes/test_responses.py`